lib/DxilContainer/DxilContainerAssembler.cpp

```diff
@@ -1668,6 +1685,58 @@ void hlsl::SerializeDxilContainerForModule(DxilModule *pModule,
     }
   }
 
+  DxilCompilerVersion VersionHeader = {};
+  if (bEmitVersionPart && pVersionInfo) {
+    UINT32 Major = 0, Minor = 0;
+    UINT32 Flags = 0;
+    IFT(pVersionInfo->GetVersion(&Major, &Minor));
+    IFT(pVersionInfo->GetFlags(&Flags));
+
+    VersionHeader.Major = Major;
+    VersionHeader.Minor = Minor;
+    VersionHeader.VersionFlags = Flags;
+    CComPtr<IDxcVersionInfo2> pVersionInfo2;
+    if (SUCCEEDED(pVersionInfo->QueryInterface(&pVersionInfo2))) {
+      UINT32 CommitCount = 0;
+      CComHeapPtr<char> CommitVersionHash;
+      IFT(pVersionInfo2->GetCommitInfo(&CommitCount, &CommitVersionHash));
+      size_t CommitLength = strlen(CommitVersionHash.m_pData);
+      size_t CopyLength = std::min(sizeof(VersionHeader.CommitSha), CommitLength);
+      memcpy(VersionHeader.CommitSha, CommitVersionHash.m_pData, CopyLength);
+      VersionHeader.VersionStringLength = 0;
```

Comment:
Why explicitly set this here?  I see that outer scope relies on zero initialization if we don't take this branch, so why not here?